### DISCUSSION TITLE
ID mapping services

### DISCUSSION THREAD

**Discussion by mihai-sysbio** [SysBioChalmers/Human-GEM#253] [Ideas]
There has been a lot work done regarding ID mapping. I'm wondering if [BridgeDB](https://bridgedb.github.io/) has been considered so far, given that is seems to cover a [wide range of identifiers](https://webservice.bridgedb.org/Homo%20sapiens/sourceDataSources) and that it has an [API](https://bridgedb.github.io/swagger/) that I can see useful for automated workflows:

> DrugBank
~miRBase Sequence~
**ChEBI**
**KEGG Reaction**
~Illumina~
Affy
Chemspider
EcoCyc
RefSeq
Unknown data source
KEGG Drug
UCSC Genome Browser
MACiE
GeneOntology
PDB
Rfam
MetaCyc
Uniprot-TrEMBL
~Agilent~
~Guide to Pharmacology~
PubChem-compound
Enzyme Nomenclature
**HMDB**
**Reactome**
**Rhea**
**Ensembl**
OMIM
LIPID MAPS
EPA CompTox
WikiGenes
InChIKey
ChEMBL compound
KNApSAcK
Wikidata
Entrez Gene
HGNC
UniGene
KEGG Compound
Complex Portal
CAS

I took the freedom to cross out those which did not sound useful and highlight the ids already used - please consider this list a work-in-progress and edit this message as you see fit. It's sad to see that MetaNetX is not on this list at the moment.

**Comment by haowang-bioinfo** [comment]
has anyone tested the performance of this [API](https://bridgedb.github.io/swagger/), if it's updated with the various data sources regularly?

**Reply by mihai-sysbio** [reply]
I have no experience in using it. If you're thinking of trying it out, there seems to be a Matlab package as well https://github.com/bridgedb/BridgeDb-matlab

**Reply by mihai-sysbio** [reply]
This is an overview of the connected databases: https://www.bridgedb.org/pages/system-codes.html

**Reply by haowang-bioinfo** [reply]
tested a bit with BridgeDb that doesn't appear to be as powerful as Metabolic Atlas. One can only get ID mappings within category, i.e. between genes or between reactions. While Met Atlas enables cross-category mapping, e.g. from gene id to that of metabolites and reactions (interaction partners).

**Reply by mihai-sysbio** [reply]
I was thinking BridgeDB could be used via an Action for curation purposes, e.g. to let us know of inaccuracies in the annotation in the TSV files.

**Reply by haowang-bioinfo** [reply]
then came up with an idea of mapping toward Rhea ids via BridgeDB

**Reply by mihai-sysbio** [reply]
Which id would you said should be used to obtain the mapping to Rhea?

**Reply by haowang-bioinfo** [reply]
By checking BridgeDB webservice, the following ids can be mapped to Rhea:
- Enzyme Nomenclature (E)
- EcoCyc (Eco)
- MACiE (Ma)
- KEGG Reaction (Rk)
- MetaCyc (Mc)
- Reactome (Re)

**Reply by mihai-sysbio** [reply]
The remaining question is - what are the benefits of using this in addition to MNXref?

**Reply by haowang-bioinfo** [reply]
> what are the benefits of using this

Very good question that had been explained mostly in #224 

Here I'd like reiterate some key points:
- Rhea is a curated and reliable data source
- It is based on the ChEBI compound database that has been tightly integrated with HumanGEM and Met Atlas
- Rhea has close associations with UniProt, with which functional annotation is manually curated and appears to be particularly useful in the recent GPT-assisted curation
- The mapping of Rhea ids will drive forward HumanGEM improvement as the evolving of Rhea

**Reply by haowang-bioinfo** [reply]
Besides BridgeDB, an alternative implementation (#506) is to map to Rhea ids directly via [rhea-db.org/mapping](https://www.rhea-db.org/mapping)

**Reply by haowang-bioinfo** [reply]
it might be good, and possible for now, to consistently map from Ensembl to Uniprot/Siwssport ID in the `genes.tsv` at some point.
